### PR TITLE
Add Dataflow pipeline to perform transpose/pivot on a BigQuery table.

### PR DESCRIPTION
**Pull request by sabhyankar**

### CODE REVIEW

File: examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/common/PivotInputProvider.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+      return Schema.of(extractedFields);
+    }
+
+    /** Validate that the pivot fields have the right {@link StandardSQLTypeName}. */
+    private void validatePivotFieldType() {
+      populateFieldNameToFieldMap();
+
+      for (String fieldName : pivotFieldNames()) {
+        if (!ALLOWED_PIVOT_TYPES.contains(
```
Comment (ryanmcdowell) [inline examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/common/PivotInputProvider.java]: This if-statement may be more readable if you write a method to evaluate the conditional logic and return the boolean result.

File: examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/common/PivotInputProvider.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            fieldNameToFieldMap.get(fieldName).getType().getStandardType())) {
+          throw new IllegalArgumentException(
+              "Unsupported pivot type: "
+                  + fieldNameToFieldMap.get(fieldName).getType().getStandardType()
+                  + " for field: "
+                  + fieldName);
+        }
+        if (fieldNameToFieldMap.get(fieldName).getMode() != null
```
Comment (ryanmcdowell) [inline examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/common/PivotInputProvider.java]: Same as above.

File: examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/common/PivotUtils.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  /**
+   * Helper method to merge a list of {@link Schema} objects and optionally sort them.
+   *
+   * @param schemas {@link Schema} objects to merge.
+   * @param sort {@link Boolean} flag to sort the fields by fieldName.
+   * @return merged {@link Schema} object.
+   */
+  private static Schema mergeSchemas(List<Schema> schemas, Boolean sort) {
```
Comment (ryanmcdowell) [inline examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/common/PivotUtils.java]: What if sort is null? Use the primitive type.

File: examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/transforms/TableRowPivot.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  abstract PCollectionView<Schema> pivotedSchema();
+
+  @Override
+  public PCollection<TableRow> expand(PCollection<TableRow> input) {
+
+    return input
+        .apply(
+            "Separate key fields",
+            MapElements.via(
+                new SimpleFunction<TableRow, KV<TableRow, TableRow>>() {
```
Comment (ryanmcdowell) [inline examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/transforms/TableRowPivot.java]: Can we write this as a lambda instead? 

```java
MapElements.via(input -> {
  // Multi-statement logic here.
});
```

File: examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/transforms/TableRowPivot.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+                                .findAny()
+                                .get();
+
+                        List<TableRow> rowsToMerge =
+                            StreamSupport.stream(context.element().getValue().spliterator(), false)
+                                .map(KV::getValue)
+                                .collect(Collectors.toList());
+
+                        TableRow mergedRow = PivotUtils.mergeTableRows(rowsToMerge, pivotSchema);
+
```
Comment (ryanmcdowell) [inline examples/dataflow-bigquery-transpose/src/main/java/com/google/cloud/pso/transforms/TableRowPivot.java]: Remove newline

**Comment by sabhyankar** [comment]
Thanks for the review @ryanmcdowell . I have refactored the code per the review suggestions.

**Comment by ryanmcdowell** [comment]
LGTM

**Pull request merged**